enterprise/models.py

```diff
@@ -322,6 +334,56 @@ def __repr__(self):
         return self.__str__()
 
 
+@python_2_unicode_compatible
+class EnterpriseCustomerIdentityProvider(TimeStampedModel):
+    """
+    EnterpriseCustomerIdentityProvider is a One to One relationship between Enterprise Customer and Identity Provider.
+
+    There should be a link between an enterprise customer and its Identity Provider. This relationship has
+    following constraints
+        1. An enterprise customer may or may not have an identity provider.
+        2. An enterprise customer can not have more than one identity providers.
+        3. Enterprise customer site should match with identity provider's site. (i.e. same domain names)
+
+    Fields:
+        enterprise_customer (ForeignKey[EnterpriseCustomer]): enterprise customer
+        provider_id (:class:`django.db.models.PositiveIntegerField`): positive integer id of identity provider.
+        provider_name (:class:`django.db.models.CharField`): Human friendly name of the identity provider.
+    """
+
+    enterprise_customer = models.OneToOneField(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        related_name="enterprise_customer_identity_provider"
+    )
+    provider_id = models.SlugField(
+        null=False,
+        blank=False,
+        unique=True,
+        help_text="Slug field containing a unique identifier for the identity provider.",
+    )
+    provider_name = models.CharField(
```

Comment:
Since `EnterpriseCustomer` already has a `name` field, and the providers each have a `name` stored in their `ProviderConfig` subclass, why do we need this additional `provider_name` field? Can you just use

```
@property
def provider_name(self):
    return get_identity_provider(self.provider_id).name
```
instead?